internal/cloudapi/v2/handler.go

```diff
@@ -1233,7 +1233,7 @@ func (h *apiHandlers) postCloneComposeImpl(ctx echo.Context, id string) error {
 					Ami:          options.Ami,
 					SourceRegion: options.Region,
 					TargetRegion: img.Region,
-					TargetName:   fmt.Sprintf("composer-api-%s", uuid.New().String()),
+					TargetName:   fmt.Sprintf("composer-api-%s", finalJob),
```

Comment:
This only change the clones (so when copying the image over to different regions). 

In general I'm not at all opposed to it but you'll probably need to pick the osbuild job because in koji jobs the final job is the koji-finalize job, which depends on up to 4 osbuild jobs, and they can't all have the same name.